src/websocket/websocket_secure_endpoint.cpp

```diff
@@ -0,0 +1,83 @@
+#include "./websocket_secure_endpoint.h"
+#include "../logging.h"
+
+namespace sensr
+{   
+  WebSocketSecureEndPoint::WebSocketSecureEndPoint(const std::string& cert_path) : 
+    WebSocketEndPointBase(),
+    cert_path_(cert_path)
+  {
+    Init(endpoint_);
+  }
+
+  WebSocketSecureEndPoint::~WebSocketSecureEndPoint() {
+    Fin(endpoint_);
+  }
+
+  bool WebSocketSecureEndPoint::Connect(const std::string &uri,
+                 const WebSocketEndPointBase::MsgReceiver& func,
+                 const WebSocketEndPointBase::ErrorReceiver& err_func) {
+    
+    if (!connection_hdl_.expired()) {
+      INFO_LOG(uri + " is already connected.");
+      return true;
+    }
+    bool ret = false;
+    try {
+      if (!cert_path_.empty()) { 
+        endpoint_.set_tls_init_handler(std::bind(
+          &WebSocketSecureEndPoint::OnTLSInit,
+          this,
+          std::placeholders::_1));
+      } else {
+        ERROR_LOG("> Certificate file path is empty.");
+        return false;
```

Comment:
Maybe we can make use of exceptions instead of these. 
```suggestion
        throw std::exception("> Certificate file path is empty.");
```